src/xtgeo/grid3d/_grdecl_format.py

```diff
@@ -15,6 +17,43 @@
 logger = null_logger(__name__)
 
 
+def run_length_encoding(arr: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Perform run-length encoding on a 1D NumPy array.
+
+    Run-length encoding is a data compression technique that represents
+    consecutive repeated values as a single value and a count.
+
+    Parameters:
+    -----------
+    arr : np.ndarray
+        A 1D NumPy array to be encoded.
+
+    Returns:
+    --------
+    tuple[np.ndarray, np.ndarray]
+        A tuple containing two 1D NumPy arrays:
+        - counts: An array of counts representing the number of consecutive occurrences
+                  of each unique value.
+        - values: An array of the unique values corresponding to the counts.
+
+    Example:
+    --------
```

Comment:
I think this is the numpy docstring style. XTGeo uses the google docstring format, so nice if you can change (although this is a private function)
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
Good point